tests/l3_test.c

```diff
@@ -0,0 +1,185 @@
+{
+    return (((logtype < 0) || (logtype >= L3_LOGTYPE_MAX))
+             ? L3_logtype_name[L3_LOG_UNDEF]
+             : L3_logtype_name[logtype]);
+}
+
+
+L3_STATIC_ASSERT((L3_ARRAY_LEN(L3_logtype_name) == L3_LOGTYPE_MAX),
+                  "Incorrect size of array L3_logtype_name[]");
+
+static FILE *  l3_test_log_fh = NULL;   // L3_LOG_FPRINTF: Opened by fopen()
+
+static int     l3_log_fd = -1;     // L3_LOG_WRITE: Opened by open()
+
+/**
+ * ****************************************************************************
+ * Initialize L3's logging sub-system to use write() to named `path`.
+ */
+static int
+l3_init_write(const char *path)
+{
+    if (!path) {
+        fprintf(stderr, "%s: Invalid arg 'path'=%p\n", __func__, path);
+        return -1;
+    }
+
+    l3_log_fd = open(path, (O_RDWR | O_CREAT | O_TRUNC), 0666);
+    if (l3_log_fd == -1) {
+        fprintf(stderr, "%s: Error opening log-file at '%p'. Error=%d\n",
+                __func__, path, errno);
+        return -1;
+    }
+    printf("Initialized write() logging to '%s'\n", path);
+    return 0;
+}
+
+/**
+ * ****************************************************************************
+ * Initialize L3's logging sub-system to use fprintf() to named `path`.
+ */
+static int
+l3_init_fprintf(const char *path)
+{
+    if (!path) {
+        fprintf(stderr, "%s: Invalid arg 'path'=%p\n", __func__, path);
+        return -1;
+    }
+
+    l3_test_log_fh = fopen(path, "w+");
+    if (!l3_test_log_fh) {
+        fprintf(stderr, "%s: Error opening log-file at '%p'. errno=%d\n",
+                __func__, path, errno);
+        perror("fopen failed");
+        return -1;
+    }
+    printf("Initialized fprintf() logging to '%s'\n", path);
+    return 0;
+}
+
+/**
+ * ****************************************************************************
+ * l3_test_log_init() - Initialize L3-logging sub-system, selecting the type of
+ * logging to be performed.
+ * ****************************************************************************
+ */
+int
+l3_test_log_init(const l3_test_log_t logtype, const char *path)
+{
+    int rv = 0;
+    switch (logtype) {
+      case L3_LOG_MMAP:         // L3_LOG_DEFAULT:
+        rv = l3_init(path);
+        break;
+
+      case L3_LOG_FPRINTF:
+        rv = l3_init_fprintf(path);
+        break;
+
+      case L3_LOG_WRITE:
+      case L3_LOG_WRITE_MSG:
+        rv = l3_init_write(path);
+        break;
+
+      default:
+        printf("Unsupported L3-logging type=%d\n", logtype);
+        return -1;
+    }
+    return rv;
+}
+
+/**
+ * ****************************************************************************
+ * l3_test_log_deinit() - De-initialize L3-logging sub-system, selecting the type
+ * of logging that was being performed. Finalize the file and close its handle.
+ * ****************************************************************************
+ */
+int
+l3_test_log_deinit(const l3_test_log_t logtype)
```

Comment:
Same above comments apply to this renamed `deinit()` interface as well.

Reply:
As above.